### PR TITLE
Feature frontend/466 coach view

### PR DESCRIPTION
**Pull request by ruvmello**
Coach view

### CODE REVIEW

**Comment by ruvmello** [comment]
Please also tell me of some other things can't be accessed by a coach, or some things need to be hidden for them.

**Comment by ruvmello** [comment]
Please also tell me of some other things can't be accessed by a coach, or some things need to be hidden for them.

File: frontend/src/components/edition/editionRowComponent.tsx [hunk 1]
```diff
@@ -60,29 +62,33 @@ export function EditionRowComponent(props: EditionProps) {
...
-                    >
-                        <Image alt="" src={"/resources/delete.svg"} width="15" height="15" />
-                    </a>
+                    {currentUserIsAdmin && (
+                        <a
+                            style={{ padding: "1rem", cursor: "pointer" }}
+                            href={transformer(
+                                applicationPaths.editionBase +
+                                    "/" +
+                                    extractIdFromEditionUrl(edition._links.self.href)
```
Comment (jitsedesmet) [inline frontend/src/components/edition/editionRowComponent.tsx:71]: Is this different form just the self.href?

File: frontend/src/components/util/forbiddenCoachRoutes.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    const isCurrentUserAdmin = useCurrentAdminUser();
+    const router = useRouter();
+
+    useEffect(() => {
+        if (isCurrentUserAdmin !== undefined && !isCurrentUserAdmin) {
+            if (pathIsForbiddenForCoach(router.asPath)) {
+                router.push("/" + applicationPaths.error).catch(console.log);
+            }
+        }
+    }, [isCurrentUserAdmin, router]);
```
Comment (jitsedesmet) [inline frontend/src/components/util/forbiddenCoachRoutes.ts:17]: You don't want to be dependent on router. It changes every render and will call this function way to much. Just be dependent on router.push and router.asPath

File: frontend/src/hooks/useCurrentUser.ts [hunk 3]
```diff
@@ -19,3 +19,11 @@ export function useCurrentUser(
     }
     return { user: data };
 }
+
+/**
+ * Hook returning if the current user is an admin or not.
+ */
+export function useCurrentAdminUser(): undefined | boolean {
+    const { user: user } = useCurrentUser();
+    return user === undefined ? undefined : user.userRole === UserRole.admin;
```
Comment (jitsedesmet) [inline frontend/src/hooks/useCurrentUser.ts:28]: Why not use the `?.` syntax anymore?

File: frontend/test/integration/edition/editionList.test.tsx [hunk 4]
```diff
@@ -37,8 +44,16 @@ describe("EditionList", () => {
...
         const edition: IEdition = getBaseActiveEdition("3", "edition 1");
-        render(<EditionRowComponent key={edition.name} edition={edition} />);
+        render(makeCacheFree(() => <EditionRowComponent key={edition.name} edition={edition} />));
 
+        await waitFor(() => expect(mockAxios.get).toHaveBeenCalled());
+
+        const responseUser: AxiosResponse = getBaseOkResponse(getBaseUser("5", UserRole.admin, true));
+        await waitFor(() => {
+            act(() => mockAxios.mockResponseFor({ url: apiPaths.ownUser }, responseUser));
+        });
```
Comment (jitsedesmet) [inline frontend/test/integration/edition/editionList.test.tsx:54]: Shoud we use the new function here?

File: frontend/test/integration/projectList.test.tsx [hunk 5]
```diff
@@ -18,9 +19,14 @@ afterEach(() => {
...
 describe("Project", () => {
     describe("ProjectList header and button", () => {
-        it("Should have the 'New project'-button", () => {
+        it("Should have the 'New project'-button", async () => {
             render(makeCacheFree(ProjectList));
-            expect(screen.getByTestId("new-project-button")).toBeInTheDocument();
+
+            await waitFor(() => expect(mockAxios.get).toHaveBeenCalled());
+            const responseUser: AxiosResponse = getBaseOkResponse(getBaseUser("5", UserRole.admin, true));
+            act(() => mockAxios.mockResponseFor({ url: apiPaths.ownUser }, responseUser));
```
Comment (jitsedesmet) [inline frontend/test/integration/projectList.test.tsx:27]: Should we use the new function here?

**Review by jitsedesmet — COMMENTED** [review]
Seems very good already. Good work Ruben! only some minor thing

File: frontend/src/components/edition/editionRowComponent.tsx (hunk 1 above)
Comment (ruvmello) [reply frontend/src/components/edition/editionRowComponent.tsx:71]: There is no edition.self.href? If that's what you mean?

Comment (jitsedesmet) [reply frontend/src/components/edition/editionRowComponent.tsx:71]: Oh, this is a frontend path, my bad!

**Review by jitsedesmet — APPROVED** [review]

**Pull request merged**